<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!--
         This node merges all the point cloud data of the primo sensor suite
     -->

    <node ns="merge0" pkg="primo_stereo" type="pcl_merge" name="merge0" output="screen">
        <param name="channel0"          value="/stereo0/vxl/output"/>
        <param name="channel0_frame"    value="stereo0_link"/>
        <param name="channel1"          value="/stereo1/vxl/output"/>
        <param name="channel1_frame"    value="stereo1_link"/>
    </node>

    <node ns="merge1" pkg="primo_stereo" type="pcl_merge" name="merge1" output="screen">
        <param name="channel0"          value="/merge0/output"/>
        <param name="channel0_frame"    value="stereo1_link"/>
        <param name="channel1"          value="/stereo2/vxl/output"/>
        <param name="channel1_frame"    value="stereo2_link"/>
    </node>

    <node ns="merge2" pkg="primo_stereo" type="pcl_merge" name="merge1" output="screen">
        <param name="channel0"          value="/merge1/output"/>
        <param name="channel0_frame"    value="stereo2_link"/>
        <param name="channel1"          value="/stereo3/vxl/output"/>
        <param name="channel1_frame"    value="zed_left_camera"/>
    </node>
</launch>

